app/navigation/navigation_modal.js

```diff
@@ -24,16 +24,20 @@ export default class NavigationModal extends PureComponent {
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.show === nextProps.show) {
+        const nextRoutesLength = nextProps.children.props.navigationState.routes.length;
+        const currentRoutesLength = this.state.children ? this.state.children.props.navigationState.routes.length : 0;
+        const routesDidChange = nextRoutesLength !== currentRoutesLength;
```

Comment:
Oh I see. Why are we copying children below when we store it in the state? Why can't we just copy the reference like we did before?

And if we do end up needing to do a deep equality check, we've got `deep-equal` that we're using elsewhere.